internal/langserver/handlers/initialize.go

```diff
@@ -39,6 +40,9 @@ func (svc *service) Initialize(ctx context.Context, params lsp.InitializeParams)
 	expClientCaps := lsp.ExperimentalClientCapabilities(clientCaps.Experimental)
 
 	svc.server = jrpc2.ServerFromContext(ctx)
+	svc.logger = logging.NewLspLogger(&logging.LspLogger{
+		Context: ctx,
```

Comment:
```suggestion
		Server: svc.server,
```
As per other comment this would make for a cleaner contract.